### ISSUE TITLE
Wrong number of arguments in warden call in 1.13.19

### ISSUE THREAD

**Issue by JasonBarnabe** [rmosolgo/graphql-ruby#4602]
**Describe the bug**

In 1.13.19, [Schema#get_field](https://github.com/rmosolgo/graphql-ruby/blob/0545517b38f735551dde9fbfa4501daab8f4a818/lib/graphql/schema/warden.rb#L138)'s signature is `get_field(parent_type, field_name)`; however [Warden#get_field](https://github.com/rmosolgo/graphql-ruby/blob/0545517b38f735551dde9fbfa4501daab8f4a818/lib/graphql/schema/warden.rb#L141) is calling it as `@schema.get_field(type, f_name, @context)`, resulting in `ArgumentError: wrong number of arguments (given 3, expected 2)`.

**Versions**

`graphql` version: 1.13.19
`rails` (or other framework): 7.0.72
other applicable versions (`graphql-batch`, etc): graphql-batch 0.3.1

**Comment by rmosolgo** [comment]
Hi! Thanks for reporting this, yes, it looks like it would raise an error. Do you mind sharing the stack trace when you encountered it? I wonder why my test suite doesn't cover it ...

**Comment by JasonBarnabe** [comment]
```
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/schema.rb:505:in `get_field'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/schema/warden.rb:141:in `block (2 levels) in get_field'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/schema/warden.rb:338:in `block in read_through'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/schema/warden.rb:150:in `get_field'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/forwardable-1.3.3/lib/forwardable.rb:240:in `get_field'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/rules/fields_will_merge.rb:343:in `block in find_fields_and_fragments'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/rules/fields_will_merge.rb:340:in `each'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/rules/fields_will_merge.rb:340:in `find_fields_and_fragments'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/rules/fields_will_merge.rb:333:in `fields_and_fragments_from_selection'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/rules/fields_will_merge.rb:61:in `conflicts_within_selection_set'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/rules/fields_will_merge.rb:25:in `block in on_operation_definition'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/rules/fields_will_merge.rb:52:in `setting_errors'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/rules/fields_will_merge.rb:25:in `on_operation_definition'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/rules/operation_names_are_valid.rb:12:in `on_operation_definition'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/rules/unique_directives_per_location.rb:27:in `block (2 levels) in <module:UniqueDirectivesPerLocation>'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/internal_representation/rewrite.rb:68:in `block in on_operation_definition'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/internal_representation/rewrite.rb:92:in `push_root_node'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/internal_representation/rewrite.rb:68:in `on_operation_definition'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/base_visitor.rb:76:in `on_operation_definition'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/language/visitor.rb:79:in `public_send'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/language/visitor.rb:79:in `visit_node'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/language/visitor.rb:174:in `on_node_with_modifications'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/language/visitor.rb:102:in `block in on_abstract_node'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/language/visitor.rb:101:in `each'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/language/visitor.rb:101:in `on_abstract_node'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/language/visitor.rb:123:in `on_document'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/definition_dependencies.rb:38:in `on_document'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/rules/variables_are_used_and_defined.rb:79:in `on_document'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/rules/fragment_spreads_are_possible.rb:26:in `on_document'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/rules/fragments_are_used.rb:6:in `on_document'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/rules/fragments_are_finite.rb:6:in `on_document'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/rules/fragment_names_are_unique.rb:17:in `on_document'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/rules/operation_names_are_valid.rb:16:in `on_document'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/rules/no_definitions_are_present.rb:34:in `on_document'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/language/visitor.rb:79:in `public_send'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/language/visitor.rb:79:in `visit_node'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/language/visitor.rb:174:in `on_node_with_modifications'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/language/visitor.rb:68:in `visit'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/validator.rb:57:in `block (3 levels) in validate'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/validator.rb:44:in `catch'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/validator.rb:44:in `block (2 levels) in validate'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/timeout-0.4.0/lib/timeout.rb:170:in `timeout'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/validator.rb:43:in `block in validate'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/tracing.rb:81:in `call_tracers'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/tracing.rb:83:in `block in call_tracers'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/tracing/platform_tracing.rb:30:in `block in trace'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/tracing/new_relic_tracing.rb:34:in `block in platform_trace'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/newrelic_rpm-9.0.0/lib/new_relic/agent/method_tracer_helpers.rb:37:in `block in trace_execution_scoped'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/newrelic_rpm-9.0.0/lib/new_relic/agent/tracer.rb:356:in `capture_segment_error'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/newrelic_rpm-9.0.0/lib/new_relic/agent/method_tracer_helpers.rb:37:in `trace_execution_scoped'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/tracing/new_relic_tracing.rb:33:in `platform_trace'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/tracing/platform_tracing.rb:29:in `trace'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/tracing.rb:83:in `call_tracers'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/tracing.rb:67:in `trace'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/static_validation/validator.rb:30:in `validate'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/query/validation_pipeline.rb:74:in `ensure_has_validated'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/query/validation_pipeline.rb:46:in `internal_representation'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/forwardable-1.3.3/lib/forwardable.rb:240:in `internal_representation'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/query.rb:242:in `irep_selection'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/schema/member/instrumentation.rb:21:in `before_query'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/execution/instrumentation.rb:58:in `public_send'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/execution/instrumentation.rb:58:in `block in call_hooks'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/execution/instrumentation.rb:57:in `each'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/execution/instrumentation.rb:57:in `call_hooks'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/execution/instrumentation.rb:44:in `each_query_call_hooks'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/execution/instrumentation.rb:27:in `block in apply_instrumenters'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/execution/instrumentation.rb:72:in `call_hooks'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/execution/instrumentation.rb:26:in `apply_instrumenters'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/execution/multiplex.rb:190:in `instrument_and_analyze'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/execution/multiplex.rb:62:in `block in run_queries'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/tracing.rb:81:in `call_tracers'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/tracing.rb:83:in `block in call_tracers'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/tracing/platform_tracing.rb:30:in `block in trace'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/tracing/new_relic_tracing.rb:34:in `block in platform_trace'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/newrelic_rpm-9.0.0/lib/new_relic/agent/method_tracer_helpers.rb:37:in `block in trace_execution_scoped'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/newrelic_rpm-9.0.0/lib/new_relic/agent/tracer.rb:356:in `capture_segment_error'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/newrelic_rpm-9.0.0/lib/new_relic/agent/method_tracer_helpers.rb:37:in `trace_execution_scoped'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/tracing/new_relic_tracing.rb:33:in `platform_trace'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/tracing/platform_tracing.rb:29:in `trace'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/tracing.rb:83:in `call_tracers'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/tracing.rb:67:in `trace'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/execution/multiplex.rb:60:in `run_queries'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/execution/multiplex.rb:50:in `run_all'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/schema.rb:1768:in `multiplex'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/graphql-1.13.19/lib/graphql/schema.rb:1739:in `execute'
/www/Simplero/test/graph/schema_test_concerns.rb:12:in `execute_query'
/www/Simplero/test/graph/types/node_type_test_concerns.rb:24:in `fetch_node'
/www/Simplero/test/graph/types/node_type_test_concerns.rb:51:in `assert_node_accessibility'
/www/Simplero/test/graph/types/nav_list_link_type_test.rb:14:in `block in <class:NavListLinkTypeTest>'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest/test.rb:94:in `block (3 levels) in run'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest/test.rb:191:in `capture_exceptions'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest/test.rb:89:in `block (2 levels) in run'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest.rb:296:in `time_it'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest/test.rb:88:in `block in run'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest.rb:405:in `on_signal'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest/test.rb:239:in `with_info_handler'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest/test.rb:87:in `run'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest.rb:1091:in `run_one_method'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest.rb:371:in `run_one_method'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest.rb:358:in `block (2 levels) in run'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest.rb:357:in `each'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest.rb:357:in `block in run'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest.rb:405:in `on_signal'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest.rb:392:in `with_info_handler'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest.rb:356:in `run'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/railties-7.0.7.2/lib/rails/test_unit/line_filtering.rb:10:in `run'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest.rb:182:in `block in __run'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest.rb:182:in `map'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest.rb:182:in `__run'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest.rb:159:in `run'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/minitest-5.19.0/lib/minitest.rb:83:in `block in autorun'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/activesupport-7.0.7.2/lib/active_support/fork_tracker.rb:18:in `fork'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/activesupport-7.0.7.2/lib/active_support/fork_tracker.rb:18:in `fork'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:85:in `require'
/home/jason/.rbenv/versions/2.7.6/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:85:in `require'
```

**Comment by rmosolgo** [comment]
I just released 1.13.20 with a fix for this issue -- please let me know if you run into any trouble on that new version!

**Comment by JasonBarnabe** [comment]
Thanks, I think I may be past this error, but now I've run into another - filed #4677.
